Open warnings

```
/home/admin/workspace/iris/build/_deps/rawterm-src/rawterm/text.cpp:63:49: warning: implicit conversion changes signedness: 'std::size_t' (aka 'unsigned long') to 'difference_type' (aka 'long') [-Wsign-conver
sion]
   63 |             if (std::regex_search(str.begin() + pos, str.end(), match, ansi_escape_code) &&
      |                                               ~ ^~~
/home/admin/workspace/iris/build/_deps/rawterm-src/rawterm/text.cpp:66:30: warning: implicit conversion changes signedness: 'difference_type' (aka 'long') to 'std::size_t' (aka 'unsigned long') [-Wsign-conver
sion]
   66 |                 pos += match.length();
      |                     ~~ ~~~~~~^~~~~~~~
2 warnings generated.
```

C++23, clang 19,1.7

--- a/examples/raw_escapes.cpp
+++ b/examples/raw_escapes.cpp
@@ -24,9 +24,9 @@ int main() {
         }
 
         std::string code;
-        for (int i = 0; i < ret; ++i) {
+        for (std::size_t i = 0; i < static_cast<std::size_t>(ret); ++i) {
             std::stringstream ss;
-            ss << std::hex << "\\x" << static_cast<unsigned char>(seq[i]);
+            ss << std::hex << "\\x" << static_cast<unsigned char>(seq.at(i));
             code += ss.str();
         }
 

--- a/rawterm/text.cpp
+++ b/rawterm/text.cpp
@@ -56,19 +56,20 @@ namespace rawterm {
         std::regex ansi_escape_code("\x1b\\[[0-9;]*[A-Za-z]");
         std::smatch match;
         int visible_index = 0;
-        unsigned int pos = 0;
+        std::size_t pos = 0;
 
         while (pos < str.length()) {
             // Check if there's an ANSI escape code at the current position
-            if (std::regex_search(str.begin() + pos, str.end(), match, ansi_escape_code) &&
+            long longPos = static_cast<long>(pos);
+            if (std::regex_search(str.begin() + longPos, str.end(), match, ansi_escape_code) &&
                 match.position() == 0) {
                 // Move position past the ANSI escape code
-                pos += match.length();
+                pos += static_cast<std::size_t>(match.length());
             } else {
                 // If the current visible index matches the target index, return the
                 // character
                 if (visible_index == index) {
-                    return str[pos];
+                    return str.at(pos);
                 }
                 // Move to the next character
                 ++visible_index;

